src/Bookings.js

```diff
@@ -1,18 +1,61 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import Search from "./Search.js";
-// import SearchResults from "./SearchResults.js";
-// import FakeBookings from "./data/fakeBookings.json";
+import SearchResults from "./SearchResults.js";
+//import FakeBookings from "./data/fakeBookings.json";
 
 const Bookings = () => {
+  const [bookings, setBookings] = useState([]);
+  const [filteredBookings, setFilteredBookings] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    fetch("https://cyf-react.glitch.me")
+      .then(res => {
+      if(res.status >= 400) {
+        throw new Error("Server responds with error!");
+      } else {
+        return res.json()
+      }})
+      .then(data => {
+        setBookings(data);
+        setIsLoading(false);
+    },
+    err => {
+      setError(err)
+      setIsLoading(false);
+    });
+  }, []);
+
+  useEffect(()=> {
+    setFilteredBookings(bookings)
+  }, [bookings])
+
+
   const search = searchVal => {
-    console.info("TO DO!", searchVal);
+   if(searchVal) {
+     const filteredBookings = bookings.filter(booking => {
+       const {firstName, surname} = booking;
+       const preparedSearchVal = searchVal.toLowerCase();
+       const preparedFirstName = firstName.toLowerCase();
+       const preparedSurname = surname.toLowerCase();
+       return preparedFirstName.includes(preparedSearchVal) || preparedSurname.includes(preparedSearchVal)
+     });
+     setFilteredBookings(filteredBookings);
+   } else {
+     setFilteredBookings(bookings);
```

Comment:
Nicely done, everything that follows has nothing to do with functionality as they are the exact same but want to talk about coding style. Do take these with a pinch of salt as styling varies from developer to developer, but worth sharing as it's always nice to know someone else's style.

I, personally, find that having too many variables can sometimes make the code harder to follow. So for example there are quite a few temporary variables within your filter that are reassigned. Sometimes that makes it easier to read but sometimes it can make it harder and is a balancing act to be aware of.

The second comment is that as your else branch is the case where you do nothing to the original bookings, it can be done in a different way to having two different calls to the setter. 
```js
const search = (searchVal) => {
  let searchQuery = searchVal.toLowerCase();
  let bookingResults = bookings;
  
  if (searchQuery) {
    bookingResults = bookings.filter({ firstName, surname } => {
      return firstName.toLowerCase().includes(searchQuery) || surname.toLowerCase().includes(searchQuery);
    });
  }
  
  setFilteredBookings(bookingResults)
}
```